### ISSUE TITLE
Geometry_Engine: Method correcting Arcs

### ISSUE THREAD

**Issue by LMarkowski** [BHoM/BHoM_Engine#1241]
#### Description:

Some Arcs from Rhino have replaced angles. E.g Instead of having StartAngle = 0 and EndAngle = 1 they have StartAngle = -1 and EndAngle = 0. Is it justified by anything?
Some of my methods base on that an arc has a StartAngle = 0 so I just wrote simple check correcting it:
`if(arc.StartAngle != 0)`
`{`
`arc.CoordinateSystem = arc.CoordinateSystem.Rotate(arc.CoordinateSystem.Origin, arc.CoordinateSystem.Z, arc.StartAngle);`
`arc.EndAngle = -arc.StartAngle;`
`arc.StartAngle = 0;`
`}`

It works like this:
![image](https://user-images.githubusercontent.com/48125514/66128740-c6b15b80-e5ee-11e9-870f-ab780d370739.png)
Geometrically those two arcs are identical.
Do you think it is worth adding to the engine?

**Comment by LMarkowski** [comment]
@pawelbaran  @IsakNaslundBh  @FraserGreenroyd  what do you think?

**Comment by LMarkowski** [comment]
Another thought:
Do we actually allow working on Arcs not having one of the angles = 0? If so, it could be a method to rearrange Arcs with messy angles.

**Comment by pawelbaran** [comment]
I would say the `Arc` should work for any values where `EndAngle > StartAngle && EndAngle - StartAngle <= Math.PI * 2` (or even less strict). However, having a method that would reset the coordinate system so that the StartAngle == 0 would be nice - can imagine this being helpful in some cases.

**Comment by IsakNaslundBh** [comment]
Not sure I agree, but do not feel to strongly about this. I think it would be fine to have startAngle != 0 as well as StartAngle > EndAngle as well as negatives to some extent. For example, working in plan could be nice to not have to redefine the coordinate system between different arcs etc.

But again, not to fuzzed either way.

**Comment by pawelbaran** [comment]
Yeah, sounds like the best option would be to allow any `StartAngle` and `EndAngle` values + have a method that would reset the CS of the arc so that `StartAngle == 0 && EndAngle > StartAngle` (can imagine this being helpful e.g. for interop/convert purposes.

**Comment by al-fisher** [comment]
Yes - agreed with that summary too. Allow any value - and have a modify engine method to create equivalent with new coordinate system and values.
